qhana_plugin_registry/db/models/templates.py

```diff
@@ -267,34 +238,30 @@ class TemplateTab(IdMixin, ExistsMixin, NameDescriptionMixin):
     def plugin_filter(self) -> dict:
         return json.loads(self.filter_string)
 
-    @classmethod
-    def get_by_name(cls, name: str) -> "Optional[TemplateTab]":
-        q = select(cls).filter(cls.name == name)
+    def get_by_name(
+        template: UiTemplate, name: str, location: Optional[str] = None
+    ) -> "Optional[TemplateTab]":
+        q = (
+            select(TemplateTab)
+            .filter(TemplateTab.template == template)
+            .filter(TemplateTab.name == name)
+        )
+        if location is not None:
+            q = q.filter(TemplateTab.location == location)
         found_tab: Optional[TemplateTab] = DB.session.execute(q).scalar_one_or_none()
         return found_tab
 
     @classmethod
-    def get_or_create(cls, tab: dict) -> "TemplateTab":
-        found_tab: Optional[TemplateTab] = cls.get_by_name(tab["name"])
+    def get_or_create(cls, template: UiTemplate, **kwargs) -> "TemplateTab":
+        from qhana_plugin_registry.tasks.plugin_filter import evaluate_plugin_filter
```

Comment:
Inline imports should be avoided if possible. (You could try if just importing the `plugin_filter` and then accessing `plugin_filter.evaluate_plugi_filter` via the module is enough to circuimvent circular imports in this case.
(However in this case an inline import is not too bad as this will only be used a few times at startup...)

Reply:
Unfortunately, importing the module (`qhana_plugin_registry.tasks.plugin_filter`) doesn't work.

<details>
<summary>Error log</summary>

```
Unable to load celery application.
While trying to load the module qhana_plugin_registry.celery_worker:CELERY the following error occurred:
Traceback (most recent call last):
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/.venv/lib/python3.10/site-packages/celery/bin/celery.py", line 57, in convert
    return find_app(value)
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/.venv/lib/python3.10/site-packages/celery/app/utils.py", line 384, in find_app
    sym = symbol_by_name(app, imp=imp)
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/.venv/lib/python3.10/site-packages/kombu/utils/imports.py", line 56, in symbol_by_name
    module = imp(module_name, package=package, **kwargs)
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/.venv/lib/python3.10/site-packages/celery/utils/imports.py", line 105, in import_from_cwd
    return imp(module, package=package)
  File "/home/user/anaconda3/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/__init__.py", line 31, in <module>
    from . import api, babel, celery, db, licenses
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/api/__init__.py", line 20, in <module>
    from .models.generators.type_map import populate_metadata
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/api/models/__init__.py", line 17, in <module>
    from . import generators  # noqa
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/api/models/generators/__init__.py", line 18, in <module>
    from . import root  # noqa
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/api/models/generators/root.py", line 49, in <module>
    from ....db.models.templates import UiTemplate
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/db/models/templates.py", line 28, in <module>
    from ...tasks import plugin_filter
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/tasks/__init__.py", line 18, in <module>
    from .plugin_discovery import purge_plugins, start_plugin_discovery
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/tasks/plugin_discovery.py", line 34, in <module>
    from .plugin_filter import update_plugin_lists
  File "/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/tasks/plugin_filter.py", line 23, in <module>
    from ..db.models.templates import TemplateTab
ImportError: cannot import name 'TemplateTab' from partially initialized module 'qhana_plugin_registry.db.models.templates' (most likely due to a circular import) (/mnt/e/Temp/hiwi/qhana-plugin-registry/qhana_plugin_registry/db/models/templates.py)
```
</details>